libs/agno/agno/agent/agent.py

```diff
@@ -739,6 +765,13 @@ def _run(
                     self.run_response.tools = model_response.tool_calls
                 else:
                     self.run_response.tools.extend(model_response.tool_calls)
+                
+                # For Reasoning/Thinking/Knowledge Tools update reasoning_content in RunResponse
+                for tool_call in model_response.tool_calls:
```

Comment:
We should have this with the tool_call_completed, so we can update the reasoning content bit by bit right?

Reply:
@dirkbrnd this is for the `non-stream` part, there is not `tool_call_completed` event for non streaming right. correct me if i'm wrong